src/main/java/org/sonar/plugins/jacoco/KotlinFileLocator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SonarQube JaCoCo Plugin
+ * Copyright (C) 2018-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.jacoco;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+public class KotlinFileLocator {
+  private static final Logger LOGGER = Loggers.get(KotlinFileLocator.class);
+  private static final String SHEBANG_LINE = "#!.*+\n";
+  private static final String DELIMITED_COMMENT = "/\\*[^(*/)]*+\\*/";
```

Comment:
Could it be that this regex doesn't quite do what was intended? Take this test:
```
    InputFile inputFile = new TestInputFileBuilder("module", "src/main/java/org/sonar/test/File.kt")
      .setContents(
        "/* " + System.lineSeparator() +
          " * package a.b " + System.lineSeparator() +
          "*/" + System.lineSeparator() +
          "package a       " + System.lineSeparator() +
          "    .`b    b`    " + System.lineSeparator() +
          ".c   "
      )
      .setCharset(Charset.defaultCharset())
      .build();
    KotlinFileLocator kotlinFileLocator = new KotlinFileLocator(Stream.of(inputFile));

    assertThat(kotlinFileLocator.getInputFile("a/b", "File.kt")).isNull();
```

It will fail because the multi-line comment is not detected as such. In the regex, is the intention of `[^(*/)]*+` to match the input as long as we don't encounter a `*/`? Currently, this is not the case, it will instead match any character that is not `(`, `)`, `*`, or `/`. 

Unless I am missing something here, I believe this is why we currently fail in the provided test case.

Reply:
you're right, that should be lookahead